lib/db.c

```diff
@@ -927,8 +927,6 @@ grn_column_create_similar_internal(grn_ctx *ctx,
                 "%s base column's source doesn't exist: %u",
                 tag,
                 base_source_column_id);
-            grn_obj_remove(ctx, column);
-            column = NULL;
```

Comment:
We need not execute `grm_obj_remove`  at this line because it is also executed at line 968 ( before fixed ).

We should not substitute `NULL` for `column because  `grm_obj_remove` ( at line 968 ) can not handle null pointer.

Reply:
The comment above is based on an assumption that `GRN_OBJ_FIN(ctx, &base_source_ids)` does not change ctx->rc to `GRN_SUCCESS`, so if the assumption is incorrect, I consider another modification.

Reply:
Generally, `GRN_OBJ_FIN()` resets` ctx->rc`.
But it's not occurred in code between `GRN_API_ENTER`/`GRN_API_RETURN`.